internal/linker/cache.go

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2022, The Garble Authors.
+// See LICENSE for licensing information.
+
+package linker
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+)
+
+const (
+	cacheDirName   = ".garble"
+	versionExt     = ".version"
+	garbleCacheDir = "GARBLE_CACHE_DIR"
+)
+
+func cachePath() string {
+	var cacheDir string
+	if val, ok := os.LookupEnv(garbleCacheDir); ok {
+		cacheDir = val
+	} else {
+		userCacheDir, err := os.UserCacheDir()
+		if err != nil {
+			panic(fmt.Errorf("cannot retreive user cache directory: %v", err))
+		}
+		cacheDir = userCacheDir
+	}
+	linkerBin := filepath.Join(cacheDir, cacheDirName, "link")
+	if runtime.GOOS == "windows" {
+		linkerBin += ".exe"
+	}
+	return linkerBin
+}
+
+func getCurrentVersion(goVersion string) []byte {
```

Comment:
nit: would be a lot easier if you use `string` for the version strings everywhere, and only convert to `[]byte` when reading or writing from disk. they are plaintext, after all, and not binary data.